### PR TITLE
Remove renamed/archived starters

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Part of #3113 

## Development notes
- I also removed the mention to `astro-iris`. This starter was renamed about 2 years ago, so I think it's fair to remove this mention now. 
- I already updated some docs where it was easy to replace the mention of `pandas-iris` with `spaceflights-pandas`, but as noted in #3160 and #3161 more doc updates need to be done. 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ tasks:
       make sign-off
       pip install -e /workspace/kedro[test]
       cd /workspace
-      yes project | kedro new -s pandas-iris --checkout main
+      yes project | kedro new -s spaceflights-pandas --checkout main
```
Comment (astrojuanlu) [inline .gitpod.yml:10]: Can we do

```suggestion
      kedro new --name project -s spaceflights-pandas --checkout main
```

already?

**Review by astrojuanlu — APPROVED** [review]
LGTM!

File: .gitpod.yml (hunk 1 above)
Comment (merelcht) [reply .gitpod.yml:10]: Ah yes I think that should work!

**Review by stichbury — APPROVED** [review]
LGTM! I'm moving the `starters.md` file and splitting it into two ("use a starter" and "create a starter") in a separate change so there will be a merge to do but I'll pick up your change and bring into mine.

**Pull request merged**